src/frontend/screens/ManualGpsScreen/index.tsx

```diff
@@ -0,0 +1,231 @@
+import {
+  View,
+  ScrollView,
+  StyleSheet,
+  ToastAndroid,
+  KeyboardAvoidingView,
+} from 'react-native';
+import {
+  FormattedMessage,
+  MessageDescriptor,
+  defineMessages,
+  useIntl,
+} from 'react-intl';
+import {NativeStackNavigationOptions} from '@react-navigation/native-stack';
+
+import {useDraftObservation} from '../../hooks/useDraftObservation';
+import {
+  usePersistedSettings,
+  usePersistedSettingsAction,
+} from '../../hooks/persistedState/usePersistedSettings';
+import {BLACK} from '../../lib/styles';
+import {IconButton} from '../../sharedComponents/IconButton';
+import {SaveIcon} from '../../sharedComponents/icons';
+import {Select} from '../../sharedComponents/Select';
+import {Text} from '../../sharedComponents/Text';
+import {
+  type NativeRootNavigationProps,
+  type CoordinateFormat,
+} from '../../sharedTypes';
+
+import {type ConvertedCoordinateData} from './shared';
+import {DdForm} from './DdForm';
+import {DmsForm} from './DmsForm';
+import {UtmForm} from './UtmForm';
+import {usePersistedDraftObservation} from '../../hooks/persistedState/usePersistedDraftObservation';
+
+const m = defineMessages({
+  title: {
+    id: 'screens.ManualGpsScreen.title',
+    defaultMessage: 'Enter coordinates',
+    description: 'title of manual GPS screen',
+  },
+  coordinateFormat: {
+    id: 'screens.ManualGpsScreen.coordinateFormat',
+    defaultMessage: 'Coordinate Format',
+  },
+  decimalDegrees: {
+    id: 'screens.ManualGpsScreen.decimalDegrees',
+    defaultMessage: 'Decimal Degrees (DD)',
+  },
+  degreesMinutesSeconds: {
+    id: 'screens.ManualGpsScreen.degreesMinutesSeconds',
+    defaultMessage: 'Degrees/Minutes/Seconds (DMS)',
+  },
+  universalTransverseMercator: {
+    id: 'screens.ManualGpsScreen.universalTransverseMercator',
+    defaultMessage: 'Universal Transverse Mercator (UTM)',
+  },
+});
+
+export const ManualGpsScreen = ({
+  navigation,
+}: NativeRootNavigationProps<'ManualGpsScreen'>) => {
+  const {formatMessage: t} = useIntl();
+
+  const [convertedData, setConvertedData] =
+    React.useState<ConvertedCoordinateData>({});
+
+  const ENTRY_FORMAT_OPTIONS = React.useMemo(
+    () => [
+      {label: t(m.decimalDegrees), value: 'dd'},
+      {label: t(m.degreesMinutesSeconds), value: 'dms'},
+      {label: t(m.universalTransverseMercator), value: 'utm'},
+    ],
+    [t],
+  );
+
+  const observationValue = usePersistedDraftObservation(
+    observationValueSelector,
+  );
+
+  const entryCoordinateFormat = usePersistedSettings(
+    entryCoordinateFormatSelector,
+  );
+
+  const {setManualCoordinateEntryFormat} = usePersistedSettingsAction();
+  const {updateObservationPosition} = useDraftObservation();
+
+  React.useLayoutEffect(() => {
+    function handleSavePress() {
+      if (convertedData.error) {
+        return ToastAndroid.showWithGravity(
+          convertedData.error.message,
+          ToastAndroid.LONG,
+          ToastAndroid.TOP,
+        );
+      }
```

Comment:
I think we should also check if `lat` and `lon` are valid here. I know that we attempted to do that in each component BUT having one last check will avoid the issue that I found in UTM form

Reply:
Addressed via e5057536eaf1ad4944d87886a98d303c7b2ea4ef